src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -381,16 +447,38 @@ const JupyterFrameManager = ({ onClose, frameRef }) => {
   return null
 }
 
-const NotebookEditorFrame = ({ mode, notebookName, workspace: { workspace: { namespace, name, bucketName } }, cluster: { clusterName, clusterUrl, status } }) => {
+const NotebookEditorFrame = ({ mode, notebookName, workspace: { workspace: { namespace, name, bucketName } }, cluster: { clusterName, clusterUrl, status, labels } }) => {
   console.assert(status === 'Running', 'Expected notebook runtime to be running')
   const frameRef = useRef()
+  const signal = useCancellation()
   const [busy, setBusy] = useState(false)
+  const [localized, setLocalized] = useState(false)
```

Comment:
Rather than having two independent code paths through this one component, what about extracting the special case into its own component, so that we can just decide up above which one to render?